packages/gdl-frontend/components/LanguageList.js

```diff
@@ -30,56 +32,121 @@ type Props = {
-            </ListItemIcon>
-            <ListItemText inset>
-              <Trans>
-                <SrOnly>Selected: </SrOnly>
-                {selectedLanguage.name}
-              </Trans>
-            </ListItemText>
-          </ListItem>
-          <Divider />
-        </>
-      )}
-      {filteredLanguages.map(l => (
-        <LanguageItem
-          key={l.code}
-          language={l}
-          linkProps={linkProps}
-          onSelectLanguage={onSelectLanguage}
+    this.state = {
+      searchResult: filteredLanguages
+    };
+  }
+
+  searchLanguage = (
+    value: SyntheticInputEvent<EventTarget>,
+    filteredLanguages: Array<Language>
+  ) => {
+    const input = value.target.value.toLowerCase();
+
+    const result = filteredLanguages.filter(element =>
+      element.name.toLowerCase().includes(input)
+    );
+    this.setState({ searchResult: result });
+  };
+
+  getSelectedLangauge = () =>
+    this.props.languages.find(l => l.code === this.props.selectedLanguageCode);
+
+  // Remove selected language so it wont display again as a selectable language
+  getFilteredLanguages = (selectedLanguage: ?Language) => {
+    const { languages } = this.props;
+    return selectedLanguage
+      ? languages.filter(l => l !== selectedLanguage)
+      : languages;
+  };
+
+  render() {
+    const { onSelectLanguage, linkProps } = this.props;
+    const { searchResult } = this.state;
+
+    const selectedLanguage = this.getSelectedLangauge();
+    const filteredLanguages = this.getFilteredLanguages(selectedLanguage);
+    const noResult = searchResult.length === 0;
+
+    return (
+      <List
+        component="div"
+        className={styles.visibleScrollbar}
+        subheader={
+          <ListSubheader component="div">
+            <Trans>Choose book language</Trans>
+          </ListSubheader>
+        }
+      >
+        <TextField
+          placeholder="Search here"
+          className={styles.textfield}
+          onChange={values => this.searchLanguage(values, filteredLanguages)}
         />
-      ))}
-    </List>
-  );
+
+        {selectedLanguage && (
+          <>
+            <ListItem>
+              <ListItemIcon>
+                <CheckIcon css={{ color: colors.base.green }} />
+              </ListItemIcon>
+              <ListItemText inset>
+                <Trans>
+                  <SrOnly>Selected: </SrOnly>
+                  {selectedLanguage.name}
+                </Trans>
+              </ListItemText>
+            </ListItem>
+            <Divider />
+          </>
+        )}
+        {noResult ? (
+          <NoLanguageItem />
+        ) : (
+          searchResult.map(l => (
+            <LanguageItem
+              key={l.code}
+              language={l}
+              linkProps={linkProps}
+              onSelectLanguage={onSelectLanguage}
+            />
+          ))
+        )}
+      </List>
+    );
+  }
+}
+
+const styles = {
+  visibleScrollbar: css`
+    overflow-y: scroll;
+    width: 300px;
```

Comment:
When searching, the list changes and renders it widths according to the longest string. I sat the width to 300px to make it consistent.

Reply:
Changed to dynamically set width based on initial width when opened.